app/views/card_grants/_spending_instructions.html.erb

```diff
@@ -7,51 +7,107 @@
+                  <%= stripe_card.formatted_card_number %>
+                <% end %>
+              <% else %>
+                <%= stripe_card.hidden_card_number_with_last_four %>
+              <% end %>
+              <%= link_to card_grant_path(card_grant, show_details: (!@show_card_details).to_s), data: { turbo: true } do %>
+                <%= inline_icon @show_card_details ? "view-fill" : "view", class: "align-middle" %>
+              <% end %>
+            <% else %>
+              <%= stripe_card.hidden_card_number_with_last_four %>
+              <div class="right tooltipped tooltipped--w" aria-label="Hidden to protect security">
+                <%= inline_icon "private", size: 32, class: "muted" %>
+              </div>
+            <% end %>
+          </div>
+        </div>
 
-        <p class="fs-mask">
+        <div class="fs-mask">
           <strong>CVC</strong>
-          <% if card_grant.user == current_user %>
-            <%= copy_to_clipboard(stripe_card.cvc, class: "w-fit", tooltip_direction: "e") %>
-          <% else %>
-            <%= stripe_card.hidden_cvc %>
-          <% end %>
-        </p>
+          <div class="flex justify-between font-mono items-center">
+            <% if stripe_card.user == current_user && stripe_card.virtual? %>
+              <% if @show_card_details %>
+                <%= copy_to_clipboard(stripe_card.cvc, class: "w-fit", tooltip_direction: "e") do %>
+                  <%= stripe_card.cvc %>
+                <% end %>
+              <% else %>
+                <%= stripe_card.hidden_cvc %>
+              <% end %>
+            <% else %>
+              <%= stripe_card.hidden_cvc %>
+              <div class="right tooltipped tooltipped--w" aria-label="Hidden to protect security">
+                <%= inline_icon "private", size: 32, class: "muted" %>
+              </div>
+            <% end %>
+          </div>
+        </div>
 
-        <p class="fs-mask">
+        <% if stripe_card.cash_withdrawal_enabled? %>
+          <div class="fs-mask" data-controller="pin" data-pin-card-id-value="<%= stripe_card.id %>" data-pin-stripe-id-value="<%= stripe_card.stripe_id %>" data-pin-stripe-api-key-value="<%= StripeService.publishable_key %>">
+            <strong>PIN</strong>
+            <div class="flex justify-between font-mono items-center">
+              <span id="card-pin">••••</span>
+              <%= inline_icon @show_card_details ? "view-fill" : "view", class: "align-middle", "data-action": "click->pin#view" %>
+            </div>
+          </div>
+        <% end %>
+
+        <div class="fs-mask">
           <strong>Expiration date</strong>
           <%= copy_to_clipboard(render_short_exp_date(stripe_card), class: "w-fit", tooltip_direction: "e") { render_exp_date stripe_card } %>
-        </p>
+        </div>
+
+        <div class="fs-mask">
+          <strong class="inline-flex items-center" style="gap: 4px">
+            Address line 1
```

Comment:
Sorry this is such a nit - but if we don't have a line 2 present can we just call this "Address"?